rank/src/services/order.ts

```diff
@@ -0,0 +1,80 @@
+import { TestCase, TestResult } from '../types/test'
+
+import { getClient } from './test'
+import { getTemplate } from './search-templates'
+import tests from '../tests'
+
+function checkResponseOrder(searchResponse, testCase: TestCase): boolean {
+  const beforeHits = searchResponse.hits.hits.filter((hit) =>
+    testCase.before.includes(hit._id)
+  )
+
+  const afterHits = searchResponse.hits.hits.filter((hit) =>
+    testCase.after.includes(hit._id)
+  )
+
+  const pass = beforeHits.every((beforeHit) => {
+    return afterHits.every((afterHit) => {
+      return beforeHit._score > afterHit._score
+    })
+  })
```

Comment:
This is fine because `beforeHits` and `afterHits` are small, but it's O(N * M) in the size of the two lists which could get thorny if you had large test cases.

Another way to structure this would be:

```typescript
const afterScores = afterHits.map((hit) => hit._score);
const beforeScores = beforeHits.map((hit) => hit._score);

return Math.min(afterScores) > Math.max(beforeScores);
```

which is only O(N + M) because you iterate through each list twice.

I don't think it's a big deal and you don't need to change it here – the majority of the rank runtime will be spent waiting for network requests – but I note for interest.